middleware/http/baggage_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2021 The OpenZipkin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http_test
+
+import (
+	"net"
+	"net/http"
+	"testing"
+
+	"github.com/openzipkin/zipkin-go"
+	"github.com/openzipkin/zipkin-go/idgenerator"
+	zipkinhttp "github.com/openzipkin/zipkin-go/middleware/http"
+	"github.com/openzipkin/zipkin-go/propagation/baggage"
+)
+
+const ReqID = "X-Request-Id"
+
+func TestHTTPBaggage(t *testing.T) {
```

Comment:
I think the test is good but it is quite hard to follow what it is doing. I think it would be helpful to get better var names and also abstract the prelude of creating server, overwriting the handler and start the listener into a private function with a descriptive name.